lib/Translation/Translator.php

```diff
@@ -254,6 +254,15 @@ public function lazyInitialize($domain, $locale)
         }
     }
 
+    /**
+    * Resets the initialization of a specific catalogue 
+    */
+    public function resetInitialization($domain, $locale)
+    {
+        $cacheKey = 'translation_data_' . md5($domain . '_' . $locale);
```

Comment:
I think it is better to refactor cachekey generation to a separate method which then also called in `lazyInitialize()`